File: core/src/com/mygdx/game/characters/State/ClimbingDown.java
package com.mygdx.game.characters.State;

import com.badlogic.gdx.math.Vector3;
import com.mygdx.game.characters.GameCharacter;
import com.mygdx.game.core.NodeHolder;

/**
 * Created by al on 5/17/2016.
 */
public class ClimbingDown extends State {
    public ClimbingDown(GameCharacter character) {
        super(character);
    }

    @Override
    public void stand() {character.state = new Stand(character);}

    @Override
    public void runLeft() {}

    @Override
    public void runRight() {}

    @Override
    public void walkLeft() {}

    @Override
    public void walkRight() {}

    @Override
    public void climbUp() {character.state = new Stand(character);}

    @Override
    public void climbDown() {
        Vector3 sub = new Vector3(0,.3f,0);
        sub.rotate(NodeHolder.rotAxis,character.globalRot);
        character.pos.add(sub);
    }
}

File: web/src/characters/state/ClimbingDown.ts
import { State } from './State';
import { Stand } from './Stand';

const CLIMB_SPEED = 0.3;

export class ClimbingDown extends State {
  stand(): void {
    this.character.fsm = new Stand(this.character);
  }
  runLeft(): void {}
  runRight(): void {}
  walkLeft(): void {}
  walkRight(): void {}
  climbUp(): void {
    this.character.fsm = new Stand(this.character);
  }
  climbDown(): void {
    this.character.addLogicalPos(0, -CLIMB_SPEED);
  }
}
